server/src/main/java/org/opensearch/wlm/stats/QueryGroupStats.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.stats;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.search.ResourceType;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * {
+ *     "queryGroupID": {
+ *          "completions": 1233234234,
+ *          "rejections": 1243545,
+ *          "failures": 97,
+ *          "CPU": { "current_usage": 49.6, "cancellation": 432 },
+ *          "MEMORY": { "current_usage": 39.6, "cancellation": 42 }
+ *     },
+ *     ...
+ *     ...
+ * }
+ */
+public class QueryGroupStats implements ToXContentObject, Writeable {
+    private final Map<String, QueryGroupStatsHolder> stats;
+
+    public QueryGroupStats(Map<String, QueryGroupStatsHolder> stats) {
+        this.stats = stats;
+    }
+
+    public QueryGroupStats(StreamInput in) throws IOException {
+        stats = in.readMap(StreamInput::readString, QueryGroupStatsHolder::new);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeMap(stats, StreamOutput::writeString, QueryGroupStatsHolder::writeTo);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        for (Map.Entry<String, QueryGroupStatsHolder> queryGroupStats : stats.entrySet()) {
+            builder.startObject(queryGroupStats.getKey());
+            queryGroupStats.getValue().toXContent(builder, params);
+            builder.endObject();
+        }
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        QueryGroupStats that = (QueryGroupStats) o;
+        return Objects.equals(stats, that.stats);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(stats);
+    }
+
+    /**
+     * This is a stats holder object which will hold the data for a query group at a point in time
+     * the instance will only be created on demand through stats api
+     */
+    public static class QueryGroupStatsHolder implements ToXContentObject, Writeable {
+        public static final String COMPLETIONS = "completions";
+        public static final String REJECTIONS = "rejections";
```

Comment:
Should we also have cancellations at the QueryGroup level?

Reply:
Yes, I will add it. We are tracking live cancellations in the state but forgot to add those in here.